web/features/search/SearchContext.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  createContext,
+  useContext,
+  useState,
+  useEffect,
+  useCallback,
+} from "react";
+import { getClient } from "../../utils/sanity/sanity.server";
+
+export type SearchResultType = {
+  slug: {
+    current: string;
+  };
+  title: string;
+  availableFrom: string;
+};
+
+const SearchContext = createContext(null);
```

Comment:
Om du sender inn typen her, så slipper du å caste lenger ned.

```suggestion
const SearchContext = createContext<SearchContextType | null>(null);
```